GTScript-Parallel-model.md

```diff
@@ -296,3 +296,118 @@ with computation(...):
         if field:
             a = a[0, 1, 0] # self assignment with offset (i.e. a read with offset and write)
 ```
+
+## Off-center writes in `K`
+
+The vertical direction, `K`, is special in weather and clinate applicaitons. It is used nut just for stencil computations but also for implicit skemes and other algorthms in physical parametrization.
+This is the reason why the vertical look is kept separate from tne horizontal IJ look in the parallel model.
+
+The vertical loop can have _directions_, indicated by the iteraiton policies FORWARD, BACKWARD and PARALLEL. To implement certain skemes it is natural to allow for off-center writes in the vertical direction, since the programmer can know in which order the values are accessed.
+
+<details>
+### Simple example
+We introduce the behavior and policies in GTScript by looking at one sample code executed the in three different iterations policies.
+
+```python
+with computation(FORWARD):  # Forward computation
+    with interval(start, end):
+        a[0,0,-1] = c
+        b[0,0,1] = 2 * a
+
+with computation(BACKWARD):  # Forward computation
+    with interval(end, start):
+        a[0,0,-1] = c
+        b[0,0,1] = 2 * a
+
+with computation(PARALLEL):  # Forward computation
+    with interval(start, end):
+        a[0,0,-1] = c
+        b[0,0,1] = 2 * a
+```
+
+This code is translated in pseudo-code as:
+
+```python
+for k in range(start, end):
+    a[i:I,j:J,k-1] = c[i:I,j:J,k]
+    b[i:I,j:J,k+1] = 2 * a[i:I,j:J,k] # b contains 2 times the previous values in a before the assignment of c, since a and b are evaluated at the same k-level
+
+for k in range(end, start):
+    a[i:I,j:J,k-1] = d[i:I,j:J,k]
+    b[i:I,j:J,k+1] = 2 * a[i:I,j:J,k] # b is going to contain the 2*d except for the firt iteration point, max2.
+
+for k in shuffle(range(start, end)):
+    a[i:I,j:J,k-1] = c[i:I,j:J,k]
+    b[i:I,j:J,k+1] = 2 * a[i:I,j:J,k] # depending on the order in which k-loop is executed we can find values from a or from c
+```
+
+### Another code sample with conditionals:
+
+```python
+with computation(FORWARD):  # Forward computation
+    with interval(start, end):
+        if (x < 0):
+            a[0,0,-1] = 1
+        else
+            a[0,0,1] = 2
+
+        b = a
+```
+
+Which translates to
+
+```python
+for k in range(start, end):
+    mask[i:I, j:J] = x < 0
+    parfor ij:
+        if (mask):
+            a[0,0,-1] = 1
+        else
+            a[0,0,1] = 2 # This will be visible in the same computation
+
+    parfor ij:
+        b = a
+```
+In this case `b` will contains 2 where `x<0`, otherwise the values contained in `a` before the computation started. This is true except for the first iteration point `min1`, for which `b` contains just the previous values contained in `a`.
```

Comment:
where `x<0` **in the k-1 slice**!!!

I think we should disallow off-center writes in `if`s on fields.

Reply:
No, why? `x` is evaluated in the current slice...